tt-train/sources/ttml/CMakeLists.txt

```diff
@@ -0,0 +1,111 @@
+project(ttml)
+
+file(
+    GLOB_RECURSE SOURCES
+    LIST_DIRECTORIES true
+    *.hpp
+    *.cpp
+) # I am a bad person
+
+add_library(ttml STATIC ${SOURCES})
+
+target_include_directories(
+    ttml
+    SYSTEM
+    PUBLIC
+        # Metalium
+        $ENV{TT_METAL_HOME}
+        $ENV{TT_METAL_HOME}/tt_metal
+        $ENV{TT_METAL_HOME}/tt_metal/third_party/umd
+        $ENV{TT_METAL_HOME}/tt_metal/hw/inc/wormhole
+        $ENV{TT_METAL_HOME}/tt_metal/hw/inc/wormhole/wormhole_b0_defines
+        $ENV{TT_METAL_HOME}/tt_metal/hw/inc/
+        $ENV{TT_METAL_HOME}/tt_metal/third_party/umd/src/firmware/riscv/wormhole
+        $ENV{TT_METAL_HOME}/tt_metal/third_party/umd/device
+        # TTNN
+        $ENV{TT_METAL_HOME}/ttnn/cpp
+        $ENV{TT_METAL_HOME}/ttnn/cpp/ttnn/deprecated
+        $ENV{TT_METAL_HOME}/tt_metal/third_party/magic_enum
+        ${reflect_SOURCE_DIR}
+        ${flatbuffers_include_dir}
+        ${nng_include_dir}
+)
+target_include_directories(ttml PUBLIC ${PROJECT_SOURCE_DIR})
+target_link_directories(ttml PUBLIC $ENV{TT_METAL_HOME}/build/lib)
+find_package(Python REQUIRED Development)
+
+foreach(lib ${BoostPackages})
+    target_include_directories(ttml SYSTEM PUBLIC ${Boost${lib}_SOURCE_DIR}/include)
+endforeach()
+
+target_link_libraries(
+    ttml
+    PUBLIC
+        dl
+        z
+        pthread
+        atomic
+)
+
+target_link_libraries(
+    ttml
+    PUBLIC
```

Comment:
Redundant.

```suggestion
```

Reply:
But that `$ENV...` for ttnn???

`Metalium::TTNN`

Reply:
Latest commit uses Metalium::TTNN

Reply:
Addressed comment in full in latest commit.

Reply:
I think almost all of them could be public.